concurrency/5/worker_test.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewDynamicWorkerPool(t *testing.T) {
+	var (
+		ctx, cancel     = context.WithCancel(context.TODO())
+		maxWorkersCount = 20
+		minWorkersCount = 3
+		tasksCh         = make(chan func(), 100)
+		taskFunc        = func() { time.Sleep(time.Second * 1) }
+		wp              = NewDynamicWorkerPool(minWorkersCount, maxWorkersCount)
+	)
+
+	go wp.Start(ctx, tasksCh)
+	for i := 0; i < 100; i++ {
+		select {
+		case tasksCh <- taskFunc:
+		case <-time.After(time.Second * 4):
+			t.Fatalf("task is not executed after 4 seconds")
+		}
+	}
+
+	//---------------
+	fmt.Println("Testing 'max load test case'")
+	time.Sleep(time.Second * 2)
+	currentWorkers := atomic.LoadInt32(wp.currentWorkerCount)
+	require.EqualValues(t, maxWorkersCount, currentWorkers,
+		"there should be max number of workers %v, but got %v", maxWorkersCount, currentWorkers)
+	fmt.Println("Successfully passed")
+	//---------------
+	fmt.Println("Testing 'load has gone' test case")
```

Comment:
the same for downscale, should we define cooldown period?

Reply:
defined it in req